### ISSUE TITLE
Ability to quickly share individual meals

### ISSUE THREAD

**Issue by scottmmjackson** [davidhealey/waistline#671]
I'm using this app with my partner, and we often eat the same meal. I would like to be able to quickly select my meal and share it using standard share options.

By my reckoning, I think this involves three things:

1. A sharing format that may be very similar to the `foodlist` json schema that already exists
2. Some kind of OS hook that allows such a file to be loaded into another copy of waistline given a well-known extension
3. UI elements for exporting and importing such a file, together with a brief explanation of what to do with the file afterwards

BTW: I've pulled down the repo and taken a look at the code, I have some free time now and am happy to offer a helping hand.

**Comment by davidhealey** [comment]
Sounds like a nice idea. JSON seems like the most suitable format for this.

**Comment by scottmmjackson** [comment]
It looks like there's some way of registering file type associations: https://stackoverflow.com/a/34610913

perhaps `.mealjson`? 😜

**Comment by davidhealey** [comment]
What about making it more generic like `.waistline` and within the object having a property that identifies it as a meal/food/recipe, etc.

**Comment by scottmmjackson** [comment]
That would be very cool. Are you familiar with the JSON Schema specification? You can easily publish open standards, plus it offers some neat verification tools you can use in CI.

**Comment by EmilJunker** [comment]
This sounds like a useful idea, but I'd suggest to first implement such a feature locally (see issues #545, #546, #547) before attempting to implement the ability to share it with another device/user.

**Comment by scottmmjackson** [comment]
@EmilJunker I think that's a reasonable statement from a roadmapping perspective. I doubt that code can be reused between these two, but it seems logical to be able to copy meals internally first before adding an export+share feature, lest the export+share feature be used to make up for the lack of the former.

**Comment by scottmmjackson** [comment]
@EmilJunker I'm happy to contribute- is there a better place to coordinate than this?

**Comment by EmilJunker** [comment]
@scottmmjackson I don't think there's a better place to discuss things than right here on GitHub. Once we have a pull request open, we should also be able to review each other's code and write comments.

**Comment by scottmmjackson** [comment]
Works for me- I'll follow up when I have something to look at.

**Comment by scottmmjackson** [comment]
WIP screenshot, thinking of this as an approach
![image](https://user-images.githubusercontent.com/1735264/213936003-c1b24355-26d9-47e5-aa6e-1c186db24500.png)

**Comment by EmilJunker** [comment]
I'd suggest to put something like a `...` button next to the `+` button that opens a menu with multiple options. There we can put the quick add (currently hidden behind long-pressing the plus button), the option to copy the meal to the clipboard, and later the ability to share it with another user.

**Comment by davidhealey** [comment]
@EmilJunker Yeah I think that's a good idea and it will be nice to make the quick add more discoverable.

**Comment by scottmmjackson** [comment]
Yeah I had no idea quick add was a thing until I opened up the source code 😂

**Comment by scottmmjackson** [comment]
@EmilJunker 
"more horiz" sound acceptable to you? https://fonts.google.com/icons?selected=Material%20Symbols%20Outlined%3Amore_horiz%3AFILL%400%3Bwght%40400%3BGRAD%400%3Bopsz%4048

**Comment by mjray** [comment]
Hi all. I'd love this. Is the work in progress available anywhere, please?
